src/withStore.js

```diff
@@ -1,25 +1,39 @@
 import { Provider } from 'mobx-react'
-import React from 'react'
+import React, { PureComponent } from 'react'
 
 /**
  * withStore is a higher order component (HOC) that allow you to initialize component without wraping any provider
  * @param { { string: Store } } stores                  - Object for mapping name of store with store object. 
- * @param {Object}              options                 - Options for modifying store lifecycle
- * @param {function}      options.onInitialized   - Function which is executed when the initialization of stores is complete.
+ * @param {Object}              options                 - Options for modifying store lifecycle.
+ * @param {function}            options.onInitialized   - Function which is executed when the initialization of stores is complete.
+ * @param {function}            options.onMounted       - Function which is executed when the component is mounted.
+ * @param {function}            options.onUnMounted     - Function which is executed when the component is unmounted.
  */
 const withStore = (stores, options) => {
     if (options) {
-        const { onInitialized } = options
+        const { onInitialized, onMounted } = options
         if (onInitialized) {
             onInitialized(stores)
         }
     }
     return Component => {
-        const ComponentWithStore = props => (
-            <Provider {...stores}>
-                <Component {...props} />
-            </Provider>
-        )
+        class ComponentWithStore extends PureComponent {
+            componentDidMount() { 
+                onMounted(stores, this.props)
```

Comment:
If `onMounted` is undefined then this would break.

I recommend either

a) `onMounted && onMounted(store, this.props)` (and change onInitialized)
b) `if(onMounted){ onMounted(...) }` to be consistent with onInitialized
c) Set defaultProps on `on*` to be a noop function